src/transformStateWithClones.js

```diff
@@ -6,8 +6,38 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneHistory = [];
+  let newClone = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      newClone = Object.assign(newClone, action.extraData);
+
+      cloneHistory.push(Object.assign({}, newClone));
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete newClone[key];
+      }
+
+      cloneHistory.push(Object.assign({}, newClone));
+    }
+
+    if (action.type === 'clear') {
+      for (const key in newClone) {
+        delete newClone[key];
+      }
+
+      cloneHistory.push(Object.assign({}, newClone));
```

Comment:
use of spread operator could be better here !

Reply:
Hi Amit, swapped the Object.assign with spread operators and replaced if-statements with switch-case. Let me know if it looks good. Thanks!